mungegithub/mungers/triage-filer.go

```diff
@@ -0,0 +1,562 @@
+func (filer *TriageFiler) RequiredFeatures() []string { return []string{} }
+
+//Initialize will initialize the TriageFiler.
+//During initialization, the TriageFiler looks up the IssueCreator and sets the nextSync time.
+func (filer *TriageFiler) Initialize(config *github.Config, features *features.Features) error {
+	filer.config = config
+	filer.nextSync = time.Now().Add(time.Duration(-1) * time.Minute) //make us due for a sync
+
+	for _, munger := range GetActiveMungers() {
+		if munger.Name() == IssueCreatorName {
+			filer.creator = munger.(*IssueCreator)
+		}
+	}
+	if filer.creator == nil {
+		return fmt.Errorf("TriageFiler couldn't find the IssueCreator among active mungers.")
+	}
+
+	return nil
+}
+
+//EachLoop is called at the start of every munge loop.
+//The TriageFiler does work only if it hasn't done so in the past syncFreqMins minutes.
+func (filer *TriageFiler) EachLoop() error {
+	if time.Now().Before(filer.nextSync) {
+		return nil
+	}
+
+	if err := filer.FileIssues(); err != nil {
+		return err
+	}
+
+	filer.nextSync = time.Now().Add(time.Duration(filer.syncFreqMins) * time.Minute)
+	glog.Infof("TriageFiler will run again at %v", filer.nextSync)
+	return nil
+}
+
+//FileIssues is the main work function of the TriageFiler.  It fetches and parses cluster data,
+//then syncs the top issues to github with the IssueCreator.
+func (filer *TriageFiler) FileIssues() error {
+	rawjson, err := mungerutil.FetchWebFileContents(filer.clusterDataURL)
+	if err != nil {
+		return err
+	}
+	clusters, err := filer.loadClusters(rawjson)
+	if err != nil {
+		return err
+	}
+	topclusters := topClusters(clusters, filer.topClustersCount)
+	for _, clust := range topclusters {
+		filer.creator.Sync(clust)
+	}
+	return nil
+}
+
+//AddFlags will add any requested flags to the cobra `cmd`.
+func (filer *TriageFiler) AddFlags(cmd *cobra.Command, config *github.Config) {
+	cmd.Flags().StringVar(&filer.clusterDataURL, "clusterdataurl", "https://storage.googleapis.com/k8s-gubernator/triage/failure_data.json", "URL pointing to JSON file from triage page that describes issue clusters.")
+	cmd.Flags().IntVar(&filer.topJobsCount, "topjobscount", 3, "The maximum number of jobs to show in an issue under 'Top failing jobs'.")
+	cmd.Flags().IntVar(&filer.topTestsCount, "toptestscount", 3, "The maximum number of tests to show in an issue under 'Top failing tests'.")
+	cmd.Flags().IntVar(&filer.topClustersCount, "triagesynccount", 10, "The number of clusters to sync issues for on github.")
+	cmd.Flags().StringVar(&filer.triageURL, "triageURL", "https://go.k8s.io/triage", "The url prefix to add before cluster hash to create a link to the triage page for a specific cluster.")
+	cmd.Flags().IntVar(&filer.windowDays, "triagewindowdays", 5, "The size of the sliding time window (in days) that is used to determine which failures to consider.")
+	cmd.Flags().IntVar(&filer.syncFreqMins, "triagesyncfreq", 60, "The frequency at which to run the TriageFiler in minutes.")
```

Comment:
Would duration be a better fit here? https://golang.org/pkg/flag/#Duration

Also can you default this to once a day? I really want to minimize the number of issues we automatically file. Said this elsewhere but too few auto-filed issues is WAY better than too many -- since too many just overwhelms people and they give up, reducing the value of all issues.